src/components/connect.js

```diff
@@ -17,9 +17,32 @@ function getDisplayName(WrappedComponent) {
 // Helps track hot reloading.
 let nextVersion = 0
 
-export default function connect(mapPropsToRequestsToProps, options = {}) {
+function connectFactory(options) {
+  function connectImpl(map) {
+    return connect(map, options)
+  }
+
+  connectImpl.withFetch = function (fetch) {
+    return connectFactory(assign({}, options, { fetch: fetch }))
+  }
+
+  connectImpl.withRef = function (withRef = true) {
+    return connectFactory(assign({}, options, { withRef: withRef }))
+  }
+
+  return connectImpl
+}
+
+const defaults = {
+  fetch: window.fetch,
+  withRef: false
+}
+
+export default connectFactory(defaults)
+
+function connect(mapPropsToRequestsToProps, options = {}) {
   const finalMapPropsToRequestsToProps = mapPropsToRequestsToProps || defaultMapPropsToRequestsToProps
-  const { withRef = false } = options
+  const { fetch, withRef } = options
```

Comment:
@passcod if you move the defaults inside connect() all the tests pass
this line becomes:
```js
const { fetch=window.fetch, withRef=false } = options
```
remove the defaults outside of connect and do
```js
export default connectFactory()
```
it's something to do with window.fetch not being defined when the module is imported, but being available at the time the connect function is called, that's why inside connect() works but outside it doesn't